ex3/assignment1_matchismo/SetCardGameViewController.m

```diff
@@ -0,0 +1,107 @@
+//
+//  SetCardGameViewController.m
+//  assignment1_matchismo
+//
+//  Created by Omer Plotnik on 11/08/2020.
+//  Copyright © 2020 lightricks. All rights reserved.
+//
+
+#import "SetCardGameViewController.h"
+#import <UIKit/UIKit.h>
+
+#import "CardGameViewController.h"
+#import "SetCard.h"
+#import "SetCardDeck.h"
+
+@interface SetCardGameViewController ()
+
+@property (weak, nonatomic) IBOutlet UILabel *activityLabel;
+@property (weak, nonatomic) IBOutlet UILabel *scoreLabel;
+
+@property (strong, nonatomic) NSMutableAttributedString *currentActivityString;
+
+@end
+
+@implementation SetCardGameViewController
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    [self updateUI];
+}
+
+
+- (CardMatchingGame *)game
+{
+    CardMatchingGame* game = [super game];
+    if(game) {
+        game.matchTwo = NO;
+    }
+    return game;
+}
+
+
+-(Deck *)createDeck
+{
+    return [[SetCardDeck alloc] init];
+}
+
+
+- (void)updateUI
+{
+    for (UIButton *cardButton in self.cardButtons) {
+        NSInteger cardButtonIndex = [self.cardButtons indexOfObject:cardButton];
+        Card *card = [self.game cardAtIndex:cardButtonIndex];
+        [cardButton setAttributedTitle:[self titleForCard:card] forState:UIControlStateNormal];
+        [cardButton setBackgroundImage:[self backgroundImageForCard:card] forState:UIControlStateNormal];
+        cardButton.enabled = !card.isMatched;
+    }
+  [super updateUI];
```

Comment:
Can this be at the first line of the function?

Reply:
Done!